google/cloud/internal/oauth2_compute_engine_credentials.cc

```diff
@@ -134,6 +208,51 @@ std::string ComputeEngineCredentials::AccountEmail() const {
   return service_account_email_;
 }
 
+StatusOr<std::string> ComputeEngineCredentials::universe_domain() const {
+  // Fetch the universe domain only once.
+  if (universe_domain_retrieved_) return universe_domain_;
+  return universe_domain(Options{});
+}
+
+StatusOr<std::string> ComputeEngineCredentials::universe_domain(
+    google::cloud::Options const& options) const {
+  // Fetch the universe domain only once.
+  if (universe_domain_retrieved_) return universe_domain_;
```

Comment:
These functions may be called by multiple threads, right?  They need some form of synchronization.  Beware keeping a lock while making RPCs.